### PR TITLE
fix: cnvpytor container

### PR DESCRIPTION
**Pull request by khurrammaqbool**
### This PR:

Changed: Updated cnvpytor to v1.3.1.
Removed: conda.
Fixed: cnvpytor container failing to download references.

### Review and tests: 
- [x] Tests pass
- [x] Code review
- [x] New code is executed and covered by tests, and test approve

### CODE REVIEW

File: BALSAMIC/containers/cnvpytor/Dockerfile [hunk 1]
```diff
@@ -1,22 +1,55 @@
...
+RUN cd /opt &&\
+    git clone https://github.com/abyzovlab/CNVpytor.git &&\
+    cd /opt/CNVpytor/ &&\
+    git checkout v1.3.1 &&\
+	pip install --no-cache-dir -e . && \
+    cnvpytor -download
 
-# Clean work environment
-RUN rm -rf /${WORK_DIR:?} && conda clean --all --yes
+from python:3.11-slim
```
Comment (ivadym) [inline BALSAMIC/containers/cnvpytor/Dockerfile]:
```suggestion
FROM python:3.11-slim
```

File: BALSAMIC/containers/cnvpytor/Dockerfile [hunk 2]
```diff
@@ -1,22 +1,55 @@
...
-# Copy all project files
-COPY . /${WORK_DIR}
+RUN pip install --no-cache-dir seaborn==0.12.2
 
-RUN cd /${WORK_DIR}/BALSAMIC/containers/${CONTAINER_NAME}/ && /bin/sh ${CONTAINER_NAME}.sh ${CONTAINER_NAME}
+RUN cd /opt &&\
+    git clone https://github.com/abyzovlab/CNVpytor.git &&\
+    cd /opt/CNVpytor/ &&\
+    git checkout v1.3.1 &&\
+	pip install --no-cache-dir -e . && \
```
Comment (ivadym) [inline BALSAMIC/containers/cnvpytor/Dockerfile:32]: Do we actually want to install it in an editable mode `-e`?

File: BALSAMIC/containers/cnvpytor/Dockerfile [hunk 3]
```diff
@@ -1,22 +1,55 @@
...
+    cnvpytor -download
 
-# Clean work environment
-RUN rm -rf /${WORK_DIR:?} && conda clean --all --yes
+FROM python:3.11-slim
```
Comment (ivadym) [inline BALSAMIC/containers/cnvpytor/Dockerfile:35]: Multi-stage builds, nice ⭐

File: BALSAMIC/containers/cnvpytor/Dockerfile [hunk 4]
```diff
@@ -1,22 +1,55 @@
...
-LABEL about.documentation="https://balsamic.readthedocs.io/"
+LABEL base.image="python:3.11-slim"
+LABEL maintainer="Clinical Genomics"
+LABEL about.contact="support@clinicalgenomics.se"
+LABEL software="cnvpytor"
+LABEL software.version="1.3.1"
+LABEL about.summary="A tool for CNV discovery and genotyping from depth-of-coverage by mapped reads"
+LABEL about.home="https://github.com/abyzovlab/CNVpytor"
+LABEL about.documentation="https://github.com/abyzovlab/CNVpytor/wiki"
 LABEL about.license="MIT License (MIT)"
```
Comment (ivadym) [inline BALSAMIC/containers/cnvpytor/Dockerfile:11]: Are these labels maintained for the final base build or should you define them there?

File: BALSAMIC/containers/cnvpytor/Dockerfile [hunk 5]
```diff
@@ -1,22 +1,55 @@
...
-LABEL about.description="Bioinformatic analysis pipeline for somatic mutations in cancer"
 
-ENV PATH="/opt/conda/bin/:${PATH}"
+ENV WORK_DIR /opt
+ENV VENV ${WORK_DIR}/venv
 
-RUN apk add --no-cache bash gcc git
+RUN apt-get update && \
+     apt-get -y upgrade && \
+     apt-get -y install -y --no-install-recommends git && \
```
Comment (ivadym) [inline BALSAMIC/containers/cnvpytor/Dockerfile:18]:
```suggestion
     apt-get -y install --no-install-recommends git && \
```

File: BALSAMIC/containers/cnvpytor/Dockerfile [hunk 6]
```diff
@@ -1,22 +1,55 @@
...
+	pip install --no-cache-dir -e . && \
+    cnvpytor -download
 
-# Clean work environment
-RUN rm -rf /${WORK_DIR:?} && conda clean --all --yes
+FROM python:3.11-slim
+
+RUN apt-get update && \
+     apt-get -y upgrade && \
+     apt-get -y install -y --no-install-recommends tabix && \
```
Comment (ivadym) [inline BALSAMIC/containers/cnvpytor/Dockerfile:39]:
```suggestion
     apt-get -y install --no-install-recommends tabix && \
```

File: BALSAMIC/containers/cnvpytor/Dockerfile [hunk 7]
```diff
@@ -1,22 +1,55 @@
...
+     apt-get clean && \
+     rm -rf /var/lib/apt/lists/*
+
+ENV WORK_DIR /opt
+ENV VENV ${WORK_DIR}/venv
+
+COPY --from=builder /opt /opt
```
Comment (ivadym) [inline BALSAMIC/containers/cnvpytor/Dockerfile:46]:
```suggestion
COPY --from=builder ${WORK_DIR} ${WORK_DIR}
```

File: BALSAMIC/containers/cnvpytor/Dockerfile [hunk 8]
```diff
@@ -1,22 +1,55 @@
...
+
+ENV PATH="${VENV}/bin:$PATH"
+
+RUN adduser --disabled-password --gecos '' ubuntu &&\
```
Comment (ivadym) [inline BALSAMIC/containers/cnvpytor/Dockerfile:50]:
```suggestion
RUN adduser --disabled-password --gecos '' ubuntu && \
```

File: BALSAMIC/containers/cnvpytor/Dockerfile [hunk 9]
```diff
@@ -1,22 +1,55 @@
...
+chsh -s /bin/bash && mkdir -p /home/ubuntu
+
+USER    ubuntu
```
Comment (ivadym) [inline BALSAMIC/containers/cnvpytor/Dockerfile:53]:
```suggestion
USER ubuntu
```

**Review by ivadym — COMMENTED** [review]
Great effort fixing this container 🎖️

File: BALSAMIC/containers/cnvpytor/Dockerfile [hunk 10]
```diff
@@ -1,22 +1,55 @@
...
+ENV VENV ${WORK_DIR}/venv
+
+RUN apt-get update && \
+    apt-get -y upgrade && \
+    apt-get -y install --no-install-recommends git && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN python -m pip install --upgrade --no-cache-dir pip
+RUN python -m venv ${VENV}
```
Comment (ivadym) [inline BALSAMIC/containers/cnvpytor/Dockerfile:13]:
```suggestion
RUN python -m pip install --upgrade --no-cache-dir pip && \
    python -m venv ${VENV}
```

File: BALSAMIC/containers/cnvpytor/Dockerfile [hunk 11]
```diff
@@ -1,22 +1,55 @@
...
+ENV PATH="${VENV}/bin:$PATH"
+
+RUN pip install --no-cache-dir seaborn==0.12.2
+
+RUN cd /opt && \
```
Comment (ivadym) [inline BALSAMIC/containers/cnvpytor/Dockerfile:18]:
```suggestion
RUN cd ${WORK_DIR} && \
```

File: BALSAMIC/containers/cnvpytor/Dockerfile [hunk 12]
```diff
@@ -1,22 +1,55 @@
...
+    git clone https://github.com/abyzovlab/CNVpytor.git && \
+    cd /opt/CNVpytor/ && \
```
Comment (ivadym) [inline BALSAMIC/containers/cnvpytor/Dockerfile:20]:
```suggestion
    cd CNVpytor && \
```

File: BALSAMIC/containers/cnvpytor/Dockerfile [hunk 13]
```diff
@@ -1,22 +1,55 @@
...
-ARG CONTAINER_NAME
+COPY --from=builder ${WORK_DIR} ${WORK_DIR}
 
-# Copy all project files
-COPY . /${WORK_DIR}
+ENV PATH="${VENV}/bin:$PATH"
 
-RUN cd /${WORK_DIR}/BALSAMIC/containers/${CONTAINER_NAME}/ && /bin/sh ${CONTAINER_NAME}.sh ${CONTAINER_NAME}
+RUN adduser --disabled-password --gecos '' ubuntu && \
+chsh -s /bin/bash && mkdir -p /home/ubuntu
```
Comment (ivadym) [inline BALSAMIC/containers/cnvpytor/Dockerfile:51]:
```suggestion
RUN adduser --disabled-password --gecos '' ubuntu && \
    chsh -s /bin/bash && mkdir -p /home/ubuntu
```

**Review by ivadym — APPROVED** [review]
Amazing @khurrammaqbool! Thanks for fixing it 💯 👼

File: BALSAMIC/containers/cnvpytor/Dockerfile (hunk 10 above)
Comment (khurrammaqbool) [reply BALSAMIC/containers/cnvpytor/Dockerfile:13]: I suggest to keep RUN.

Comment (ivadym) [reply BALSAMIC/containers/cnvpytor/Dockerfile:13]: 👍

**Pull request merged**